### PR TITLE
Bluetooth: Controller: Enable Connection Subrating support in the SDC

### PR DESCRIPTION
**Pull request by weeTike**
This enables experimental support for Connection Subrating.

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig [hunk 1]
```diff
@@ -38,6 +38,7 @@ config BT_LL_SOFTDEVICE
 	select BT_HAS_HCI_VS
 	select BT_CTLR_DF_SUPPORT if HAS_HW_NRF_RADIO_DFE
 	select BT_CTLR_LE_POWER_CONTROL_SUPPORT
+	select BT_CTLR_SUBRATING_SUPPORT
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/Kconfig:41]: You will need to merge this kconfig to zephyr before merging here, also I don't think we should select it as it is not currently supported even in an experimental state

File: subsys/bluetooth/controller/hci_internal.c [hunk 2]
```diff
@@ -1528,6 +1537,13 @@ static uint8_t le_controller_cmd_put(uint8_t const * const cmd,
 									(void *)event_out_params);
 #endif
 
+#if defined(CONFIG_BT_CTLR_SUBRATING)
+	case SDC_HCI_OPCODE_CMD_LE_SET_DEFAULT_SUBRATE:
+		return sdc_hci_cmd_le_set_default_subrate((void *)cmd_params);
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/hci_internal.c:1542]: Set default subrate can only be called by a central, so probably best to guard this with CONFIG_BT_CENTRAL, same https://github.com/nrfconnect/sdk-nrf/pull/15500/files#diff-f5bf9d2614b19c06ca472417c8004daeae8125af4a59c8ec6983180c9ddae3daR613

File: subsys/bluetooth/controller/Kconfig (hunk 1 above)
Comment (weeTike) [reply subsys/bluetooth/controller/Kconfig:41]: It is necessary to select subrating support here

File: subsys/bluetooth/controller/hci_driver.c [hunk 3]
```diff
@@ -175,6 +175,12 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
 #define SDC_LE_POWER_CONTROL_MEM_SIZE 0
 #endif
 
+#if defined(BT_CTLR_SUBRATING)
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/hci_driver.c:178]:
```suggestion
#if defined(CONFIG_BT_CTLR_SUBRATING)
```

File: subsys/bluetooth/controller/hci_driver.c [hunk 4]
```diff
@@ -242,6 +248,7 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
 		      (SDC_CENTRAL_COUNT * CENTRAL_MEM_SIZE) + \
 		      (SDC_ADV_SET_MEM_SIZE) + \
 		      (SDC_LE_POWER_CONTROL_MEM_SIZE) + \
+			  (SDC_SUBRATING_MEM_SIZE) + \
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/hci_driver.c:251]: Formatting seems off

File: subsys/bluetooth/controller/Kconfig [hunk 5]
```diff
@@ -563,5 +564,9 @@ config BT_CTLR_SDC_CIS_SUBEVENT_LENGTH_US
 	  If this parameter is set to zero, the subevent length
 	  is chosen by the controller.
 
+config BT_CTLR_SUBRATING
+	bool "Subrate Connection commands [EXPERIMENTAL]"
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/Kconfig:568]:
```suggestion
	bool "LE Connection Subrating [EXPERIMENTAL]"
```

**Review by sean-madigan — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]

**Pull request merged**